independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/EndpointIndexer.java

```diff
@@ -469,7 +475,65 @@ private void validateHttpAnnotations(MethodInfo info) {
         }
     }
 
-    private boolean hasProperModifiers(MethodInfo info) {
+    public static Set<MethodInfo> findEndpoints(ResourceScanningResult scanningResult, IndexView index,
```

Comment:
Where has the logic for this method come from?

Reply:
I derived it from `org.jboss.resteasy.reactive.common.processor.EndpointIndexer#createEndpoints(org.jboss.jandex.ClassInfo, boolean)` as to keep firm link between checks performed by `EagerSecurityHandler` I need to have exactly same endpoints https://github.com/quarkusio/quarkus/blob/main/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/ResteasyReactiveProcessor.java#L584

Reply:
btw it does not use annotation store as AFAIK you can't consume annotationtransformations and produce it due to circular dependency

Reply:
> I derived it from `org.jboss.resteasy.reactive.common.processor.EndpointIndexer#createEndpoints(org.jboss.jandex.ClassInfo, boolean)` as to keep firm link between checks performed by `EagerSecurityHandler` I need to have exactly same endpoints https://github.com/quarkusio/quarkus/blob/main/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/ResteasyReactiveProcessor.java#L584

So we have a duplication in how endpoints are looked up?

Reply:
Yes. `createEndpoints` does combine look up/validation/creation of `org.jboss.resteasy.reactive.common.model.ResourceClass`. I am happy to separate look up, but please bear in mind it will require bigger changes to `createEndpoints`. Should I do it?

Reply:
Let me look at this closer and let you know.